Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -60,7 +61,13 @@ void Processor::flush_tlb(Memory::PageDirectory const*, VirtualAddress vaddr, si
 
 u32 Processor::clear_critical()
 {
-    TODO_AARCH64();
+    InterruptDisabler disabler;
+    auto prev_critical = in_critical();
+    auto& proc = current();
+    proc.m_in_critical = 0;
+    if (proc.m_in_irq == 0)
```

Comment:
This is almost identical except that `m_in_critical` is written directly instead of using `write_gs_ptr`. Maybe this function could be in Kernel/Arch/Common/Processor.cpp and it could call a per-arch helper for that one line?

(Not necessarily in this patch.)

Reply:
Yep, I agree we should definitely have some general Processor.cpp file that we can share between x86_64 and aarch64, it's on my :yaklist: :)

I'm just wondering whether it should be `Kernel/Arch/Common/Processor.cpp`, or if we just put it in the general `Kernel` directory, but we can figure that out later :)